From a31174f922504f54acf47b2088f75c37d1f2cfa1 Mon Sep 17 00:00:00 2001
From: sezero <sezero@users.sourceforge.net>
Date: Tue, 20 Mar 2018 11:30:00 +0300
Subject: [PATCH] os.h: include config.h

---
 os.h |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/os.h b/os.h
index 12ba684..69dc200 100644
--- a/os.h
+++ b/os.h
@@ -17,6 +17,10 @@
 
  ********************************************************************/
 
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
 #include <math.h>
 #include <ogg/os_types.h>
 
-- 
1.7.1

