diff -u libogg-1.3.2/configure.in~ libogg-1.3.2/configure.in
--- libogg-1.3.2/configure.in~
+++ libogg-1.3.2/configure.in
@@ -45,13 +45,13 @@
         case $host in 
         *-*-linux*)
                 DEBUG="-g -Wall -fsigned-char"
-                CFLAGS="-O20 -Wall -ffast-math -fsigned-char"
-                PROFILE="-Wall -W -pg -g -O20 -ffast-math -fsigned-char"
+                CFLAGS="-O2 -Wall -ffast-math -fsigned-char"
+                PROFILE="-Wall -W -pg -g -O2 -ffast-math -fsigned-char"
 		;;
         sparc-sun-*)
                 DEBUG="-g -Wall -fsigned-char"
-                CFLAGS="-O20 -ffast-math -fsigned-char"
-                PROFILE="-pg -g -O20 -fsigned-char" 
+                CFLAGS="-O2 -ffast-math -fsigned-char"
+                PROFILE="-pg -g -O2 -fsigned-char" 
 		;;
 	*-*-darwin*)
 		DEBUG="-fno-common -g -Wall -fsigned-char"
@@ -60,8 +60,8 @@
 		;;
         *)
                 DEBUG="-g -Wall -fsigned-char"
-                CFLAGS="-O20 -fsigned-char"
-                PROFILE="-O20 -g -pg -fsigned-char" 
+                CFLAGS="-O2 -fsigned-char"
+                PROFILE="-O2 -g -pg -fsigned-char" 
 		;;
         esac
 fi
diff -u libogg-1.3.2/configure~ libogg-1.3.2/configure~
--- libogg-1.3.2/configure~
+++ libogg-1.3.2/configure
@@ -11727,13 +11727,13 @@
         case $host in
         *-*-linux*)
                 DEBUG="-g -Wall -fsigned-char"
-                CFLAGS="-O20 -Wall -ffast-math -fsigned-char"
-                PROFILE="-Wall -W -pg -g -O20 -ffast-math -fsigned-char"
+                CFLAGS="-O2 -Wall -ffast-math -fsigned-char"
+                PROFILE="-Wall -W -pg -g -O2 -ffast-math -fsigned-char"
 		;;
         sparc-sun-*)
                 DEBUG="-g -Wall -fsigned-char"
-                CFLAGS="-O20 -ffast-math -fsigned-char"
-                PROFILE="-pg -g -O20 -fsigned-char"
+                CFLAGS="-O2 -ffast-math -fsigned-char"
+                PROFILE="-pg -g -O2 -fsigned-char"
 		;;
 	*-*-darwin*)
 		DEBUG="-fno-common -g -Wall -fsigned-char"
@@ -11742,8 +11742,8 @@
 		;;
         *)
                 DEBUG="-g -Wall -fsigned-char"
-                CFLAGS="-O20 -fsigned-char"
-                PROFILE="-O20 -g -pg -fsigned-char"
+                CFLAGS="-O2 -fsigned-char"
+                PROFILE="-O2 -g -pg -fsigned-char"
 		;;
         esac
 fi
