From 66d060c7734c9b780d0183f6565d474fc246b84b Mon Sep 17 00:00:00 2001
From: Tom Denton <tomdenton@google.com>
Date: Thu, 6 Jan 2022 09:35:12 -0800
Subject: [PATCH] Initialize non-zero test arrays.

Signed-off-by: Jean-Marc Valin <jmvalin@jmvalin.ca>
---
 celt/x86/pitch_sse4_1.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/celt/x86/pitch_sse4_1.c b/celt/x86/pitch_sse4_1.c
index 58db6c7f2..2bc578304 100644
--- a/celt/x86/pitch_sse4_1.c
+++ b/celt/x86/pitch_sse4_1.c
@@ -118,7 +118,10 @@ void xcorr_kernel_sse4_1(const opus_val16 * x, const opus_val16 * y, opus_val32
     __m128i initSum;
 
 #ifdef OPUS_CHECK_ASM
-    opus_val32 sum_c[4]={0,0,0,0};
+    opus_val32 sum_c[4];
+    for (j=0;j<4;j++) {
+      sum_c[j] = sum[j];
+    }
     xcorr_kernel_c(x, y, sum_c, len);
 #endif
 
