diff -u /dev/null libvorbisidec-1.2.1/Android.mk
--- /dev/null
+++ libvorbisidec-1.2.1/Android.mk
@@ -0,0 +1,32 @@
+LOCAL_PATH := $(call my-dir)
+
+include $(CLEAR_VARS)
+
+LOCAL_MODULE := vorbisidec
+
+LOCAL_C_INCLUDES :=
+
+LOCAL_CFLAGS := -DHAVE_ALLOCA -DVAR_ARRAYS
+
+ifeq ($(TARGET_ARCH),arm)
+	LOCAL_CFLAGS += -D_ARM_ASSEM_
+endif
+
+LOCAL_SRC_FILES += \
+    mdct.c.arm \
+    block.c \
+    window.c.arm \
+    synthesis.c \
+    info.c \
+    floor1.c.arm \
+    floor0.c.arm \
+    vorbisfile.c.arm \
+    res012.c \
+    mapping0.c \
+    registry.c \
+    codebook.c \
+    sharedbook.c.arm \
+
+LOCAL_SHARED_LIBRARIES := ogg
+
+include $(BUILD_STATIC_LIBRARY)
diff -u libvorbisidec-1.2.1/misc.h~ libvorbisidec-1.2.1/misc.h
--- libvorbisidec-1.2.1/misc.h~
+++ libvorbisidec-1.2.1/misc.h
@@ -30,6 +30,7 @@
 
 #include "asm_arm.h"
 #include <stdlib.h> /* for abs() */
+#include <endian.h>
   
 #ifndef _V_WIDE_MATH
 #define _V_WIDE_MATH
